### PR TITLE
Rename W2DImage to PlatformImage

### LINKED ISSUE #402: Documentation - load image resource doesn't compile on Windows
### Description

https://docs.microsoft.com/en-us/dotnet/maui/user-interface/graphics/paint#create-an-imagepaint-object
This section shows how to draw an image on a canvas.
Loading the image doesn't work _**(does not compile)**_ on Windows (it only works on Android (I have tried Windows and Android)) and there is not any mention of this in the article and there is not any mention of how to load the image on Windows.
Isn't MAUI supposed to be cross-platform?
Isn't there any nice and clean code to load the image cross-platform?

I can see there that build action "EmbeddedResource" must be used instead of "MauiImage" which is default.
I assume that EmbeddedResource creates something like a byte[] resource while MauiImage creates a whole set of images which have specific resolution for each platform etc.

Do I have to put a duplicate image into the resources just so that the same image can be used as a image in ImageButton and at the same time drawn on a canvas (the first one requires MauiImage and the second one EmbeddedResource)?

Thank you

Edit:
- added more info:

`error CS0234: The type or namespace name 'Platform' does not exist in the namespace 'Microsoft.Maui.Graphics' (are you missing an assembly reference?)
`

image

image

### Steps to Reproduce

Try the example on Windows

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #457: Help me! The type or namespace name 'Platform' does not exist in the namespace 'Microsoft.Maui.Graphics'
The type or namespace name 'Platform' does not exist in the namespace 'Microsoft.Maui.Graphics'

pls i cannot find Platform Class when build MAUI windows app 

code :

public void Draw(ICanvas canvas, RectF dirtyRect)
    {
        // Drawing code goes here

        IImage image;
        Assembly assembly = GetType().GetTypeInfo().Assembly;
        using (Stream stream = assembly.GetManifestResourceStream("MemeCreator.Resources.Images.dotnet_bot.svg"))
        {
            image = PlatformImage.FromStream(stream); // error here

        }

        if (image != null)
        {
            IImage newImage = image.Resize(100, 60, ResizeMode.Stretch, true);
            canvas.DrawImage(newImage, 10, 10, newImage.Width, newImage.Height);
        }
    }

Screenshot 2022-06-22 213120

### LINKED ISSUE #722: For Windows, How do I load an image resource and draw it on a GraphicsView?
I simply want to draw an image resource onto a `GraphicsView` (on Microsoft's Windows platform).  This seems like a fairly basic task.  After all, it's a "Multi-plAtform UI" right? :)

So, I fired up Google, found this documentation, and followed the sample code.  After failing to succeed, I went back to the documentation and noticed I missed this warning:

"Warning: The PlatformImage type isn't supported on Windows."

Any chance the page can be updated to provide something more useful? Perhaps, something like "On the Windows platform..." do this instead?  I'm really trying to like MAUI, but it simply doesn't seem ready for prime time yet.

I hope these early experiences don't scare other people away from it.  For my part, I'm going to wait 6 months and try again.  I was super-excited about its promise, but less so now,

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Version Independent ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Content: [Draw graphical objects - .NET MAUI](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/graphics/draw)
* Content Source: [docs/user-interface/graphics/draw.md](https://github.com/dotnet/docs-maui/blob/main/docs/user-interface/graphics/draw.md)
* Product: **dotnet-mobile**
* Technology: **dotnet-maui**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

### LINKED ISSUE #820: No way to load image on Windows
Example states that "The PlatformImage type isn't supported on Windows." and doesn't mention any way to load an image in windows.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Version Independent ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Content: [Draw graphical objects - .NET MAUI](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/graphics/draw)
* Content Source: [docs/user-interface/graphics/draw.md](https://github.com/dotnet/docs-maui/blob/main/docs/user-interface/graphics/draw.md)
* Product: **dotnet-mobile**
* Technology: **dotnet-maui**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

### LINKED ISSUE #1122: MAUI images not cross-platform on Windows?
On this page, in the "Draw an Image" section, there is a notation that `PlatformImage` is not supported on Windows.  It seems crazy to have a "**Multi-Platform** Application UI", from Microsoft, that supports all platforms **except** Microsoft Windows.
 
I've asked this before, but I was hoping, with the release of .NET 7, that this might have changed. However, it appears it is still an issue.

Am I really forced to write platform-specific code for something as simple as drawing an image on a canvas? Is there even a way to do this on Windows?

Any idea if/when there will be a mechanism for drawing images on a canvas that is **truly** multi-platform?

It seems this should be slightly embarrassing for Microsoft...that an example in their documentation has to explicitly exclude their own operating system.

For further examples of the havoc this causes, I direct you to the following page: [.NET MAUI / Graphics / Images](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/images).

The many, many notes/warnings about the incompatibilities inherent in MAUI image handling reinforce the impression that MAUI is "not ready for prime time". I honestly hope this impression, and the delays resolving it, won't negatively impact its future.

Thanks in advance for your time.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Version Independent ID: 0ca86947-204f-9c45-223d-724fb59b4938
* Content: [Draw graphical objects - .NET MAUI](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/graphics/draw?view=net-maui-7.0)
* Content Source: [docs/user-interface/graphics/draw.md](https://github.com/dotnet/docs-maui/blob/main/docs/user-interface/graphics/draw.md)
* Product: **dotnet-mobile**
* Technology: **dotnet-maui**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

### LINKED ISSUE #8509: Microsoft.Maui.Graphics.Platform error
### Description

I am trying to display photos in a NetMaui app by using the GraphicsView control.
This worked properly until the latest release of Maui, when the following error appeared:
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name 'Platform' does not exist in the namespace 'Microsoft.Maui.Graphics' (are you missing an assembly reference?)	TestMauiGraphics (net6.0-windows10.0.19041.0)	E:\Projects\NetMAUI\TestMauiGraphics\TestMauiGraphics\MainPage.xaml.cs	35	Active
The error is not shown as marked on the source by VS2022 17,3.0 Preview 2.0
I have created a sample project to show the error at https://github.com/pgmariotti/TestMauiGraphics
Please tell me what I am doing wrong...

### Steps to Reproduce

Just compile the program

### Version with bug

6.0.400 (current)

### Last version that worked well

6.0.312

### Affected platforms

Windows

### Affected platform versions

Windows 11 SDK 10.0.22000.0

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jfversluis**
_Copied details from: #9948_

### Description of Change

With this changes we can avoid this issue:
image
I have renamed not only W2DImage, but also W2Canvas and other classes to align the naming with the rest of platforms.

NOTE: This will be landing in .NET 8 because of the API changes

### Issues Fixed

Fixes #8509
Fixes https://github.com/dotnet/Microsoft.Maui.Graphics/issues/457
Fixes https://github.com/dotnet/Microsoft.Maui.Graphics/issues/402
Fixes https://github.com/dotnet/docs-maui/issues/722
Fixes https://github.com/dotnet/docs-maui/issues/820
Fixes https://github.com/dotnet/docs-maui/issues/1122

### CODE REVIEW

**Review by jsuarezruiz — COMMENTED** [review]
Pending changes in the public API `D:\a\_work\1\s\src\Core\src\Platform\Windows\GraphicsExtensions.cs(113,50): error RS0016: Symbol 'AsPoint' is not part of the declared API [D:\a\_work\1\s\src\Core\src\Core.csproj::TargetFramework=net7.0-windows10.0.19041]`

**Comment by mattleibow** [comment]
Instead of a straight rename, can we rather make it conditional. If we just rename we break all existing code.

By conditional I mean we can make the win2d nuget still using the old name, and add the new name to graphics. We can use file linking and set some defines.

This was we can make win2d ToPlatformImage and stuff convert to the new types.

This is all because breaking this now might be very bad. And we can obsolete the old name too.

File: src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -1900,10 +1900,10 @@
 ~override Microsoft.Maui.Controls.Handlers.Items.SelectableItemsViewHandler<TItemsView>.DisconnectHandler(Microsoft.UI.Xaml.Controls.ListViewBase platformView) -> void
 ~override Microsoft.Maui.Controls.Handlers.Items.StructuredItemsViewHandler<TItemsView>.Layout.get -> Microsoft.Maui.Controls.IItemsLayout
 ~override Microsoft.Maui.Controls.Handlers.Items.StructuredItemsViewHandler<TItemsView>.SelectListViewBase() -> Microsoft.UI.Xaml.Controls.ListViewBase
-~override Microsoft.Maui.Controls.Handlers.PolygonHandler.ConnectHandler(Microsoft.Maui.Graphics.Win2D.W2DGraphicsView nativeView) -> void
```
Comment (mattleibow) [inline src/Controls/src/Core/PublicAPI/net-windows/PublicAPI.Shipped.txt:1903]: We can't edit the shipped files. Only a release branch can.

**Review by mattleibow — CHANGES_REQUESTED** [review]
This is all very good but we need to keep the old win2d package and use that to redirect customers.

We can also do implicit converters if need be.

**Comment by jfversluis** [comment]
Closing this in favor of #12578

**Pull request closed without merging**